python/lsst/dax/data_generator/timingdict.py

```diff
@@ -0,0 +1,146 @@
+#!/usr/bin/env python3
+
+# Developed for the LSST Data Management System.
+# This product includes software developed by the LSST Project
+# (http://www.lsst.org).
+# See the COPYRIGHT file at the top-level directory of this distribution
+# for details of code ownership.
+#
+# This program is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with this program.  If not, see <http://www.gnu.org/licenses/>.
+
+import time
+
+
+class TimingDict:
+    """A dictionary used to store timing information
+
+    Parameters
+    ----------
+    timing_dict : dictionary
+        A dicitionary with str keys and float values. The dictionary is
+        used internally by the class. If it is None, an empty dictionary
+        is created.
+
+    Notes
+    -----
+    While this is intended to store timing information, the only hard
+    requirement is that all the keys are strings and all the values are
+    floats.
+    The key "count" is reserved for the total count.
+    """
+
+    def __init__(self, timing_dict=None):
```

Comment:
It's used by the server when importing the dictionaries from the clients. I agree about the name though, maybe self._times?

Reply:
The main issue I have is that if `__init__()` just sets the instance variable to the parameter, that breaks encapsulation and makes it harder to do something like using a `defaultdict` internally. `combine()` is doing the right thing by taking the values, not the object itself, so it might be better to either let the server create an empty instance and call combine, or call combine from inside `__init__()`.

`self._timing_dict` I think is fine inside the class, it's just when it appears outside or in the function signature that it's confusing.